themes/default/content/blog/managing-github-with-pulumi/index.md

```diff
@@ -0,0 +1,599 @@
+    12 unchanged
+
+Do you want to perform this import?  yes
+Importing (pulumi/prod)
+
+     Type                  Name                Status
+     pulumi:pulumi:Stack   team-mgmt-prod
+ =   └─ github:index:Team  animals             imported
+
+Resources:
+    = 1 imported
+    1 unchanged
+```
+
+Now that we have imported the `Team` resource, it is part of our Pulumi Stack. But it is not part of our main.go program yet. Helpfully, Pulumi Import replies with sample code on how to add the imported resource to your Pulumi program, in the language you selected for your project:
+
+```bash
+Please copy the following code into your Pulumi application. Not doing so will cause Pulumi to report that an update will happen on the next update command.
+
+Please note that the imported resources are marked as protected. To destroy them you will need to remove the `protect` option and run `pulumi update` *before* the destroy will take effect.
+
+
+package main
+
+import (
+    "github.com/pulumi/pulumi-github/sdk/v4/go/github"
+    "github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func main() {
+    pulumi.Run(func(ctx *pulumi.Context) error {
+        _, err := github.NewTeam(ctx, "animals", &github.TeamArgs{
+            CreateDefaultMaintainer: pulumi.Bool(false),
+            Description:             pulumi.String("should get imported"),
+            Name:                    pulumi.String("animals"),
+            Privacy:                 pulumi.String("closed"),
+        }, pulumi.Protect(true))
+        if err != nil {
+            return err
+        }
+        return nil
+    })
+}
+```
+
+After adding the team to `main.go` as suggested, this is the output of `pulumi preview`:
+
+```bash
+Previewing update (pulumi/prod)
+
+     Type                   Name                 Plan       Info
+     pulumi:pulumi:Stack    team-mgmt-prod
+
+Resources:
+    2 unchanged
+```
+
+Since `pulumi preview` shows no changes, we now know that our code reflects the existing infrastructure. It’s a bit funny to think about your program working well when it does nothing, but this was a huge first step in preserving existing infrastructure and ensuring all of our coworkers could continue their daily work uninterrupted!
+
+To finish up, we [unprotect the resource](https://www.pulumi.com/docs/reference/cli/pulumi_state_unprotect/):
+
+```bash
+$ pulumi state unprotect 'urn:pulumi:prod::team-mgmt::github:index/team:Team::animals'
+```
+
+## Creating Configuration
+
+Next, we import the remaining teams and refactor the code to handle multiple teams at once. This is also the time to write our organization structure into a yaml configuration file.
+
+```yaml
+---
+org: "pulumi-demo--org"
+teams:
+ - name: "animals"
+   slug: "animals"
+   description: "A team in charge of animals"
+ - name: "newteam"
+   slug: "newteam"
+   description: "I am a new team"
+```
+
+We then read this information into a Go struct with the necessary properties to pass into the Pulumi provider.
+
+```go
```

Comment:
Leaving a note here in case anyone asks: Normally, we try to publish all code in multiple languages. In this case, I don't think that directive makes sense, so leave it as is.